cohttp-lwt-unix/src/server.ml

```diff
@@ -66,7 +68,8 @@ let log_on_exn = function
 
 let create ?timeout ?backlog ?stop ?(on_exn = log_on_exn)
     ?(ctx = Net.default_ctx) ?(mode = `TCP (`Port 8080)) spec =
-  Conduit_lwt_unix.serve ?backlog ?timeout ?stop ~on_exn ~ctx:ctx.Net.ctx ~mode
-    (fun flow ic oc ->
-      let ic = Input_channel.create ic in
-      callback spec flow ic oc)
+  Lwt_eio.Promise.await_lwt
+    (Conduit_lwt_unix.serve ?backlog ?timeout ?stop ~on_exn ~ctx:ctx.Net.ctx
+       ~mode (fun flow ic oc ->
+         let ic = Input_channel.create ic in
+         Lwt.return (callback spec flow ic oc)))
```

Comment:
Anywhere you're wrapping an Eio function like this you probably want `Lwt_eio.run_eio` instead. Otherwise, if `callback` performs an effect then it will block the Lwt event loop and could prevent other Lwt threads from making progress.